Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ISISIndirectEnergyTransferWrapper.py

```diff
@@ -0,0 +1,200 @@
+                        PropertyMode, WorkspaceGroupProperty, WorkspaceProperty)
+
+
+def exists_in_ads(workspace_name):
+    return AnalysisDataService.doesExist(workspace_name)
+
+
+def get_ads_workspace(workspace_name):
+    return AnalysisDataService.retrieve(workspace_name) if exists_in_ads(workspace_name) else None
+
+
+def string_or_none(string):
+    return string if string != '' else None
+
+
+def workspace_or_none(workspace_name):
+    return get_ads_workspace(workspace_name) if workspace_name != '' else None
+
+
+def elements_or_none(elements):
+    return elements if len(elements) != 0 else None
+
+
+class ISISIndirectEnergyTransferWrapper(DataProcessorAlgorithm):
+
+    _data_files = None
+    _sum_files = None
+    _load_logs = None
+    _calibration_workspace = None
+
+    _instrument_name = None
+    _analyser = None
+    _reflection = None
+    _efixed = None
+
+    _spectra_range = None
+    _background_range = None
+    _rebin_string = None
+    _detailed_balance = None
+    _scale_factor = None
+    _fold_multiple_frames = None
+
+    _grouping_method = None
+    _grouping_workspace = None
+    _grouping_string = None
+    _grouping_map_file = None
+
+    _output_x_units = None
+    _output_workspace = None
+
+    def category(self):
+        return 'Workflow\\Inelastic;Inelastic\\Indirect'
+
+    def summary(self):
+        return 'Runs an energy transfer reduction for an inelastic indirect geometry instrument. It is a wrapper for ' \
+               'the ISISIndirectEnergyTransfer algorithm.'
+
+    def PyInit(self):
+        # Input properties
+        self.declareProperty(StringArrayProperty(name='InputFiles'),
+                             doc='Comma separated list of input files')
+
+        self.declareProperty(name='SumFiles', defaultValue=False,
+                             doc='Toggle input file summing or sequential processing')
+
+        self.declareProperty(name='LoadLogFiles', defaultValue=True,
+                             doc='Load log files when loading runs')
+
+        self.declareProperty(WorkspaceProperty('CalibrationWorkspace', '',
+                                               direction=Direction.Input,
+                                               optional=PropertyMode.Optional),
+                             doc='Workspace containing calibration data')
+
+        # Instrument configuration properties
+        self.declareProperty(name='Instrument', defaultValue='',
+                             validator=StringListValidator(['IRIS', 'OSIRIS', 'TOSCA', 'TFXA']),
+                             doc='Instrument used during run.')
+        self.declareProperty(name='Analyser', defaultValue='',
+                             validator=StringListValidator(['graphite', 'mica', 'fmica']),
+                             doc='Analyser bank used during run.')
+        self.declareProperty(name='Reflection', defaultValue='',
```

Comment:
should this and the 2 above default to one of the valid options?

Reply:
I think the defaults here should stay as is because there is no preferred value out of the valid options